src/features/fireWeather/slices/forecastsSlice.ts

```diff
@@ -0,0 +1,76 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+
+import { Forecast, getNoonForecasts, NoonForecastValue } from 'api/forecastAPI'
+import { AppThunk } from 'app/store'
+
+interface State {
+  loading: boolean
+  error: string | null
+  noonForecastsByStation: Record<number, NoonForecastValue[] | undefined>
+  noonForecasts: Record<number, {} | undefined>
+}
+
+const initialState: State = {
+  loading: false,
+  error: null,
+  noonForecastsByStation: {},
+  noonForecasts: []
+}
+
+const forecastsSlice = createSlice({
+  name: 'forecasts',
+  initialState,
+  reducers: {
+    getForecastsStart(state: State) {
+      state.loading = true
+    },
+    getForecastsFailed(state: State, action: PayloadAction<string>) {
+      state.loading = false
+      state.error = action.payload
+    },
+    getForecastsSuccess(state: State, action: PayloadAction<Forecast[]>) {
+      state.loading = false
+      state.error = null
```

Comment:
Can you put these in at the end?
I realized processing data take times, and we want the loading spinner to keep going until the processing is done